### PR TITLE
Feature/40/popup space card modal on clicking space card

### PR DESCRIPTION
**Pull request by eden-chan**

### CODE REVIEW

File: components/Spaces/Dashboard/SpaceCard.js [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+
+export default function SpaceCard(props) {
+  const theme = useTheme();
+  const { description, headCount, music } = props;
+  return (
+    <Paper className="rounded-xl">
+      <Box className="flex flex-grow justify-between ">
+        <Box color={theme.palette.primary.dark} className="text-left p-3">
+          <Typography variant="h5" align="center">
+            {props.spaceName}
```
Comment (LemonFace0309) [inline components/Spaces/Dashboard/SpaceCard.js:21]: destructure again. No one does props.spaceName. Instead.
`export default function SpaceCard({spaceName, ...}) {
  ...
  {spaceName}
  ...
}`

File: components/Spaces/Dashboard/SpaceCardModal.js [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+          <ListItemText primary={user.name} secondary={user.status} />
+        </ListItem>
+      ))}
+    </List>
+  );
+}
+
+export default function SpaceCardModal(props) {
+  const theme = useTheme();
+  const { onClose, open, children, friends, participants, hosts } = props;
```
Comment (LemonFace0309) [inline components/Spaces/Dashboard/SpaceCardModal.js:40]: destructure on line 38 instead please

File: components/Spaces/Dashboard/SpaceCardModal.js [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+  
+
+  const handleClose = () => {
+    onClose();
+  };
+
```
Comment (LemonFace0309) [inline components/Spaces/Dashboard/SpaceCardModal.js:46]: don't need it's own function. Just do `<Dialog onClose={() => onClose()} />`
onClose isn't a very good function name either. `handleClose` would be more appropriate

File: components/Spaces/Dashboard/SpaceCardModal.js [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+            <Typography variant="body1" color="textSecondary">
+              Participants
+            </Typography>
+            <UserList users={participants} />
+          </Box>
+
+          <Box className="flex flex-col p-5">
+            <Typography variant="body1" color="textSecondary">
+              Host&#40;s&#41;
```
Comment (LemonFace0309) [inline components/Spaces/Dashboard/SpaceCardModal.js:78]: is this necessary? Honestly unsure

File: components/Spaces/StudySpace/User.js [hunk 5]
```diff
@@ -0,0 +1,26 @@
+import React from "react";
+
+import { Paper, Box, Typography, Avatar } from "@material-ui/core";
+import { useTheme } from "@material-ui/core/styles";
+
+import PersonIcon from "@material-ui/icons/Person";
```
Comment (LemonFace0309) [inline components/Spaces/StudySpace/User.js:6]: these are all external packages. Just group them imo

File: pages/spaces/spaceCardDemo.js [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+  // When using ThemeProvider, use the useTheme() hook.
+  // const theme = useTheme();
+
+  const { cardData } = spaceCardTestData;
+
+  return (
+    <ThemeProvider theme={theme}>
+      <div className="flex flex-row space-x-5">
+        {cardData.map((data) => {
+          let { spaceName, description, headCount, music } = data;
```
Comment (LemonFace0309) [inline pages/spaces/spaceCardDemo.js:32]: you can just destructure on line 31 instead of line 32.

File: pages/spaces/spaceCardModalDemo.js [hunk 7]
```diff
@@ -0,0 +1,87 @@
...
+
+  const [open, setOpen] = useState(false);
+
+  const handleClickOpen = () => {
+    setOpen(true);
+  };
+
+  const handleClose = () => {
+    setOpen(false);
+  };
```
Comment (LemonFace0309) [inline pages/spaces/spaceCardModalDemo.js:45]: same thing here

File: pages/spaces/spaceCardModalDemo.js [hunk 8]
```diff
@@ -0,0 +1,87 @@
...
+
+  return (
+    <ThemeProvider theme={theme}>
+      <div className="flex flex-row space-x-5">
+        {cardData.map((data) => {
+          let { spaceName, description, headCount, music } = data;
```
Comment (LemonFace0309) [inline pages/spaces/spaceCardModalDemo.js:51]: and here

File: pages/spaces/studySpaceDemo.js [hunk 9]
```diff
@@ -0,0 +1,123 @@
...
+
+          {/* MUI grid can be an item AND a container. */}
+          <Grid item container className="col-span-8">
+            <Container className="mb-24">
+              <Grid className="grid grid-cols-6 ml-5 space-x-5 space-y-5 ">
+                <div className="mt-5 ml-5">
+                  <User />
+                </div>
+                <User /> <User /> <User /> <User /> <User />
+                <User /> <User /> <User /> <User /> <User /> <User />
```
Comment (LemonFace0309) [inline pages/spaces/studySpaceDemo.js:105]: is there a better way than this?

**Review by LemonFace0309 — CHANGES_REQUESTED** [review]
Overall, very good job. You're making a lot of progress. Fix the suggestions, add package-lock.json back, and we'll go from there

**Comment by LemonFace0309** [comment]
By the way. Press `ctrl + alt + F` to parse code with prettier. The hotkey may be different for linux. You may also need prettier installed in vscode but i'm unsure.

**Comment by LemonFace0309** [comment]
Due to how directive routing is setup, you should rename files in the `pages` folder like `study-space-demo` instead of `studySpaceDemo`

**Comment by LemonFace0309** [comment]
Also, generally in prs, it's good habit to upload photos of the ui you worked on. That way the reviewer doesn't have to open a new environment and search for it themself.

**Comment by LemonFace0309** [comment]
image
can you make this more responsive? It doesn't look natural on my viewport

File: components/Spaces/Dashboard/SpaceCardModal.js (hunk 4 above)
Comment (eden-chan) [reply components/Spaces/Dashboard/SpaceCardModal.js:78]: The box component is for the second column
image
Unless you mean the entities, which are for the parentheses in host (s)

File: components/Spaces/StudySpace/User.js (hunk 5 above)
Comment (eden-chan) [reply components/Spaces/StudySpace/User.js:6]: image
I can only group useTheme into the core package

Comment (LemonFace0309) [reply components/Spaces/StudySpace/User.js:6]: No I mean remove line 2 and line 5 lol.

File: components/Spaces/Dashboard/SpaceCardModal.js (hunk 4 above)
Comment (LemonFace0309) [reply components/Spaces/Dashboard/SpaceCardModal.js:74]: I mean the cryptic characters `Host&#40;s&#41;`

**Comment by eden-chan** [comment]
Merged into refactor/30... branch

**Pull request closed without merging**